src/v/storage/parser.cc

```diff
@@ -386,13 +391,15 @@ class copy_helper {
     ss::output_stream<char> _output;
     record_batch_transform_predicate _pred;
     model::record_batch_header _header{};
+    opt_abort_source_t _as;
 };
 
 ss::future<result<size_t>> transform_stream(
```

Comment:
I think that in the end the `abort_source` is not passed in any of the `transform_streams` calls

Reply:
it's passed in `offset_translator::copy_stream`